libs/@blockprotocol/graph/src/stdlib/subgraph/element/map-revisions.ts

```diff
@@ -0,0 +1,43 @@
+import { Vertex } from "../../../types/subgraph/vertices.js";
+
+type BaseIdToRevisions<
+  TemporalSupport extends boolean,
+  GraphElementType extends Vertex<TemporalSupport>["inner"],
+> = Record<
+  /*
+   * @todo - we _should_ be able to use `Extract<GraphElementForIdentifier<TemporalSupport, VertexId<any, any>>`
+   *   here to actually get a strong type (like `EntityId` or `BaseUri`). TypeScript seems to break on using it with a
+   *   generic though. So for now we write `string` because all of the baseId's of `VertexId` are string aliases anyway.
+   */
```

Comment:
This is actually quite frustrating and if we end up making the BaseIds template strings or branded types this will catch us out.